diff --git a/camera/provider/2.5/default/Android.bp b/camera/provider/2.5/default/Android.bp
index 9ddf65144..27bf32b43 100644
--- a/camera/provider/2.5/default/Android.bp
+++ b/camera/provider/2.5/default/Android.bp
@@ -154,7 +154,6 @@ cc_binary {
     proprietary: true,
     relative_install_path: "hw",
     srcs: ["external-service.cpp"],
-    compile_multilib: "32",
     init_rc: ["android.hardware.camera.provider@2.5-external-service.rc"],
     shared_libs: [
         "android.hardware.camera.common@1.0",
